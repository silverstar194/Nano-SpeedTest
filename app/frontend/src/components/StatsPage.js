import React, {Component} from 'react';
import Header from './Header'
import TableRow from './TableRow'
import '../styles/HomePage.css';

class StatsPage extends Component {
    render() {
        const fakeTableData = [
            {
                hash: 'abc-123',
                origin: 'SF',
                destination: 'LA',
                time: 9.3,
                amount: 12345,
                completed: 'Jan 1, 2018'
            }, {
                hash: 'zyx-321',
                origin: 'NY',
                destination: 'WI',
                time: 9.7,
                amount: 55555,
                completed: 'March 15, 2015'
            }
        ]
        return (
            <div className="StatsPage">
                <Header/>
                <h1 className="page-header text-center">Stats Page</h1>
                <table className="table">
                    <thead>
                        <tr>
                        <th scope="col">#</th>
                        <th scope="col">Hash</th>
                        <th scope="col">Origin</th>
                        <th scope="col">Destination</th>
                        <th scope="col">Elapsed Time</th>
                        <th scope="col">Amount</th>
                        <th scope="col">Completed</th>
                        </tr>
                    </thead>
                    <tbody>
                    {
                        fakeTableData.map((props, index) => {
                            props.index = index;
                            return <TableRow key={props.hash} {...props}/>
                        })
                    }
                    </tbody>
                    </table>
            </div>
        );
    }
}

export default StatsPage;