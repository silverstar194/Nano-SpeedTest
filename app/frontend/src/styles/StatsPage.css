/*source: https://www.w3schools.com/howto/howto_css_loader.asp*/
.loader {
    border: 8px solid #f3f3f3; /* Light grey */
    border-top: 8px solid #3498db; /* Blue */
    border-radius: 50%;
    width: 120px;
    height: 120px;
    animation: spin 2s linear infinite;
    padding: 10px;
}

@keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
}
/*************************************************************/
