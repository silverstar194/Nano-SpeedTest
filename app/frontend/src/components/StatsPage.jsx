import React, {Component} from 'react';
import Header from './Header';
import PropTypes from 'prop-types';
import TableRow from './TableRow';
import '../styles/StatsPage.css';
import Map from './Map';
import {connect} from 'react-redux';
import {addTransaction} from "../actions/table";
import uuid from 'uuid';

class StatsPage extends Component {
    state = {

    };

    render() {
        let cities = {};
        const {table} = this.props;
        if (table.length) {
            const transaction = table[table.length - 1]; // grab the last entry that we have loaded
            const origin = transaction['origin'];
            const originIndex = origin.indexOf(',');
            const destination = transaction['destination'];
            const destinationIndex = destination.indexOf(',');

            cities = {
                origin: {
                    name: 'Mumbai', //hard coded for now
                    coords: {
                        lat: parseFloat(origin.substring(0, originIndex)),
                        lng: parseFloat(origin.substring(originIndex + 1))
                    }
                },
                destination: {
                    name: 'Virginia', //hard coded for now
                    coords: {
                        lat: parseFloat(destination.substring(0, destinationIndex)),
                        lng: parseFloat(destination.substring(destinationIndex + 1))
                    }
                }
            };
        }
        // render the jsx
        return (
            <div className='StatsPage'>
                <Header/>
                {
                    table.length ?
                    (
                        <div>
                            <h2 className='map-header page-header text-left'>
                                Sent from {cities.origin.name} to {cities.destination.name}
                            </h2>
                            <div className='nano-container'>
                                <Map cities={cities}/>
                            </div>
                            <div className='nano-container'>
                                <table className='table'>
                                    <thead>
                                    <tr>
                                        <th scope='col'>#</th>
                                        <th scope='col'>Hash</th>
                                        <th scope='col'>Origin</th>
                                        <th scope='col'>Destination</th>
                                        <th scope='col'>Elapsed Time</th>
                                        <th scope='col'>Amount</th>
                                        <th scope='col'>Completed</th>
                                    </tr>
                                    </thead>
                                    <tbody>
                                    {
                                        table.map((props, index) => {
                                            props.index = index + 1;
                                            return <TableRow key={uuid(props.hash)} {...props}/>;
                                        })
                                    }
                                    </tbody>
                                </table>
                            </div>

                            {/* /*** only show loading if we are waiting for the data.  Else show button to rerun
                                {this.state.loading && (
                                    <div className='loading-container'>
                                        <div className='loader-container d-flex justify-content-center'>
                                            <div className='loader'></div>
                                        </div>
                                        <div>
                                            <p>Your transaction is processing. Please wait.</p>
                                        </div>
                                    </div> )
                                }  */}
                        </div>
                    )
                    : (
                        <h2>
                            No results to show
                            <p></p>
                            Please go to the Speed Test tab and run a test
                        </h2>
                    )
                }
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        table: state.table
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        onAddData(data) {
            dispatch(addTransaction(data));
        }
    };
};

StatsPage.propTypes = {
    table: PropTypes.array.isRequired,
    onAddData: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(StatsPage);