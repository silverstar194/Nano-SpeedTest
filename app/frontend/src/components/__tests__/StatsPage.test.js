import React from 'react';
import renderer from 'react-test-renderer';
import StatsPage from '../StatsPage';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter} from 'react-router-dom';
import { shallow } from 'enzyme';

const mockStore = configureStore();
let store = mockStore({
    navigation: {
        activeTab: ''
    },
    table: [
        {
            index: 0,
            hash: 'abc123',
            origin: 'SF',
            destination: 'LA',
        }
    ]
});

const props = {
	history: {}
};

it('shallow renders without crashing', () => {
    shallow(<StatsPage store={store} {...props}/>);
});

it(`renders snapshot correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter >
                <StatsPage {...props}/>
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
