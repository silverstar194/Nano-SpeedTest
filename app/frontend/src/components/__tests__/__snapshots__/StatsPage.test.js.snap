// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div
  className="StatsPage"
>
  <div
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
         
        Speed Test
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Stats"
        onClick={[Function]}
      >
         
        Statistics
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Info"
        onClick={[Function]}
      >
         
        More Info
         
      </a>
    </li>
  </div>
  <div>
    <h2
      className="map-header page-header text-left"
    >
      Sent from 
      Mumbai
       to 
      Virginia
    </h2>
    <div
      className="nano-container"
    >
      <div
        className="GoogleMapsContainer"
      >
        <div
          className="GoogleMapsElement"
        />
        <div />
      </div>
    </div>
    <div
      className="nano-container"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              #
            </th>
            <th
              scope="col"
            >
              Hash
            </th>
            <th
              scope="col"
            >
              Origin
            </th>
            <th
              scope="col"
            >
              Destination
            </th>
            <th
              scope="col"
            >
              Elapsed Time
            </th>
            <th
              scope="col"
            >
              Amount
            </th>
            <th
              scope="col"
            >
              Completed
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              scope="row"
            >
              1
            </th>
            <td>
              abc123
            </td>
            <td>
              SF
            </td>
            <td>
              LA
            </td>
            <td>
               Seconds
            </td>
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
